lib/ruby-push-notifications/mpns/mpns_notification.rb

```diff
@@ -0,0 +1,66 @@
+
+module RubyPushNotifications
+  module MPNS
+    # Encapsulates a MPNS Notification.
+    # Actually support for raw, toast, tiles notifications
+    # (http://msdn.microsoft.com/en-us/library/windowsphone/develop/hh202945)
+    #
+    class MPNSNotification
+
+      # @return [Array]. Array with the results from sending this notification
+      attr_accessor :results, :data, :device_urls
+
+      # Initializes the notification
+      #
+      # @param [Array]. Array with the receiver's device urls.
+      # @param [Hash]. Payload to send.
+      #   Toast :title => a bold message
+      #     :message => the small message
+      #     :param => a string parameter that is passed to the app
+      #   Tile :image => a new image for the tile
+      #     :count => a number to show on the tile
+      #     :title => the new title of the tile
+      #     :back_image => an image for the back of the tile
+      #     :back_title => a title on the back of the tile
+      #     :back_content => some content (text) for the back
+      #   Raw :message => the full XML message body
+      def initialize(device_urls, data)
+        @device_urls = device_urls
+        @data = data
+        @data[:type] ||= 'raw'
```

Comment:
Why you moved from symbols to strings? Symbols are the preferred way for this kind of 'constant' values